p2p/net/swarm/swarm_dial.go

```diff
@@ -337,7 +345,7 @@ func (s *Swarm) resolveAddrs(ctx context.Context, pi peer.AddrInfo) ([]ma.Multia
 	// * Otherwise, resolve it and add the results to the "to resolve" list.
 	toResolve := append([]ma.Multiaddr{}, pi.Addrs...)
 	resolved := make([]ma.Multiaddr, 0, len(pi.Addrs))
```

Comment:
I think this code would be nicer if we change the invariants to:

1. Addresses in the resolved list are all fully resolved.
2. Addresses in the toResolve list are DNS addresses that aren't fully resolved.